contracts/modules/Checkpoint/ERC20DividendCheckpoint.sol

```diff
@@ -0,0 +1,195 @@
+pragma solidity ^0.4.23;
+
+import "./ICheckpoint.sol";
+import "../../interfaces/ISecurityToken.sol";
+import "openzeppelin-solidity/contracts/math/SafeMath.sol";
+import "openzeppelin-solidity/contracts/math/Math.sol";
+import "openzeppelin-solidity/contracts/token/ERC20/ERC20.sol";
+
+contract ERC20DividendCheckpoint is ICheckpoint {
+    using SafeMath for uint256;
+
+    struct Dividend {
+      uint256 checkpointId;
+      uint256 created; // Time at which the dividend was created
+      uint256 maturity; // Time after which dividend can be claimed - set to 0 to bypass
+      uint256 expiry;  // Time until which dividend can be claimed - after this time any remaining amount can be withdrawn by issuer - set to very high value to bypass
+      address token; // Address of ERC20 token that dividend is denominated in
+      uint256 amount; // Dividend amount in base token amounts
+      uint256 claimedAmount; // Amount of dividend claimed so far
+      uint256 totalSupply; // Total supply at the associated checkpoint (avoids recalculating this)
+      bool reclaimed;
+      mapping (address => bool) claimed; // List of addresses which have claimed dividend
+    }
+
+    // List of all dividends
+    Dividend[] public dividends;
+
+    event ERC20DividendDeposited(address indexed _depositor, uint256 _checkpointId, uint256 _created, uint256 _maturity, uint256 _expiry, address _token, uint256 _amount, uint256 _totalSupply, uint256 _dividendIndex);
+    event ERC20DividendClaimed(address indexed _payee, uint256 _dividendIndex, address _token, uint256 _amount);
+    event ERC20DividendReclaimed(address indexed _claimer, uint256 _dividendIndex, address _token, uint256 _claimedAmount);
+
+    modifier validDividendIndex(uint256 _dividendIndex) {
+        require(_dividendIndex < dividends.length, "Incorrect dividend index");
+        require(now >= dividends[_dividendIndex].maturity, "Dividend maturity is in the future");
+        require(now < dividends[_dividendIndex].expiry, "Dividend expiry is in the past");
+        require(dividends[_dividendIndex].reclaimed == false, "Dividend has been reclaimed by issuer");
```

Comment:
chenge to require (!dividends[_dividendIndex].reclaimed) to keep consistency

Reply:
Done